kit/azure/networking/locals.tf

```diff
@@ -0,0 +1,81 @@
+locals {
+  diag_resource_list = var.diagnostics != null ? split("/", var.diagnostics.destination) : []
+  parsed_diag = var.diagnostics != null ? {
+    log_analytics_id   = contains(local.diag_resource_list, "Microsoft.OperationalInsights") ? var.diagnostics.destination : null
+    storage_account_id = contains(local.diag_resource_list, "Microsoft.Storage") ? var.diagnostics.destination : null
+    event_hub_auth_id  = contains(local.diag_resource_list, "Microsoft.EventHub") ? var.diagnostics.destination : null
+    metric             = var.diagnostics.metrics
+    log                = var.diagnostics.logs
+    } : {
+    log_analytics_id   = null
+    storage_account_id = null
+    event_hub_auth_id  = null
+    metric             = []
+    log                = []
+  }
+
+  default_nsg_rule = {
+    direction                                  = "Inbound"
+    access                                     = "Allow"
+    protocol                                   = "Tcp"
+    description                                = null
+    source_port_range                          = null
+    source_port_ranges                         = null
+    destination_port_range                     = null
+    destination_port_ranges                    = null
+    source_address_prefix                      = null
+    source_address_prefixes                    = null
+    source_application_security_group_ids      = null
+    destination_address_prefix                 = null
+    destination_address_prefixes               = null
+    destination_application_security_group_ids = null
+  }
```

Comment:
Q: What effect does this rule have? Source and destination ranges are set to null. So no effect?

Reply:
https://github.com/meshcloud/collie-hub/blob/3bbb0194ab0d31caf0f2f1958863a9f8b33f515b/kit/azure/networking/resources.network.tf#L250 you could use every possible value in the resource.

Reply:
That makes sense. I have one more question about the priority of rules, see separate comment.